Formula/hasura-cli.rb

```diff
@@ -0,0 +1,28 @@
+class HasuraCli < Formula
+  desc "Command-Line Interface for Hasura GraphQL Engine"
+  homepage "https://hasura.io"
+  # TODO: Add stable url when v1.0.0 is released
+  head "https://github.com/hasura/graphql-engine.git"
+
+  depends_on "go" => :build
+
+  def install
+    Dir.chdir("cli")
+    modname = "github.com/hasura/graphql-engine/cli"
+    exflags = '-extldflags "-static"'
+    ldflags = "-ldflags '-X #{modname}/version.BuildVersion=#{version} -s -w #{exflags}'"
+    File.open("build.sh", "w") do |file|
+      # Using a build file to work around an odd string interpolation issue with the single/double quotes
+      file.puts "set -eox pipefail"
+      file.puts "go mod init #{modname}"
+      file.puts "go build -o ./bin/hasura -a -v #{ldflags} ./cmd/hasura/"
+    end
+    chmod "+x", "build.sh"
+    system "./build.sh"
+    bin.install "./bin/hasura"
+  end
+
+  test do
+    system "hasura", "version"
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.